snapcraft/internal/project_loader.py

```diff
@@ -131,6 +132,15 @@ def __init__(self, project_options=None):
         snapcraft_yaml = self._expand_filesets(snapcraft_yaml)
         self.data = self._expand_env(snapcraft_yaml)
 
+        # Prevent multiple parts from having duplicate alias names
+        aliases = []
+        for app in self.data.get('apps', []):
+            app = self.data['apps'][app]
+            for alias in app.get('aliases', []):
```

Comment:
Shouldn't we show the user the alias that is duplicated?  I initially thought of doing something along these lines but opted for this approach to be able to show the duplicated alias.  I'm happy to change this if you think that not showing the duplicate alias is okay.